### PR TITLE
Add training loss dynamics exportation feature for detection task

### PR DESCRIPTION
**Pull request by vinnamkim**
### Summary
 - Ticket no. 93287
 - Add training loss dynamics exportation feature for detection task.
 - This PR only supports `CustomATSS` detector. Following up PRs will cover other models.

### How to test
I added unit tests about this change.

### Checklist

- [x] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [x] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [x] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by sungmanc — APPROVED** [review]
LGTM

File: otx/algorithms/detection/adapters/mmdet/task.py [hunk 1]
```diff
@@ -145,6 +146,10 @@ def _init_task(self, dataset: Optional[DatasetEntity] = None, export: bool = Fal
         # Update recipe with caching modules
         self._update_caching_modules(self._recipe_cfg.data)
 
+        # Loss dynamics tracking
```
Comment (jaegukhyun) [inline otx/algorithms/detection/adapters/mmdet/task.py:149]: As we decide to collect inner patch of config to configurer, this should be moved to DetectionConfigurer. 
I'd like to suggest to add enable_noisy_label_detection flag in patch_from_hyperparams function. patch_from_hyperparams is for merging config and hyperparameters. After that DetectionConfigurer should patch config when cfg.enable_noisy_label_detection is True

**Review by jaegukhyun — COMMENTED** [review]
Overall codes look good to me. Since I'm not familiar with loss dynamics, I don't closely look at the algorithm in detail. I wish I could review it after introduction meeting. And I want to suggest to move config patch parts to existed patch function and configurer.

**Review by wonjuleee — APPROVED** [review]
LGTM
